server/pulp/server/controllers/repository.py

```diff
@@ -271,6 +272,33 @@ def queue_delete(repo_id):
     return async_result
 
 
+def get_importer_by_id(object_id):
+    """
+    Get a plugin and call configuration using the document ID
+    of the repository-importer association document.
+
+    :param object_id: The document ID .
+    :type object_id: str
+    :return: A tuple of:
+        (pulp.plugins.importer.Importer, pulp.plugins.config.PluginCallConfiguration)
+    :rtype: tuple
+    :raise pulp.plugins.loader.exceptions.PluginNotFound: not found.
+    :raise
+    """
+    collection = RepoImporter.get_collection()
+    try:
+        object_id = ObjectId(object_id)
+    except InvalidId:
+        raise plugin_exceptions.PluginNotFound()
```

Comment:
I'm on the fence about this. On the one hand it might be better to let the InvalidId exception be raised because that lets the user know they aren't providing the correct input, but on the other hand the PluginNotFound exception isn't *wrong* and will still make the user go check their input.